system_tests/recreatestate_rpc_test.go

```diff
@@ -259,9 +262,166 @@ func TestRecreateStateForRPCBlockNotFoundWhileRecreating(t *testing.T) {
 	_, err = l2client.BalanceAt(ctx, GetTestAddressForAccountName(t, "User2"), new(big.Int).SetUint64(lastBlock))
 	if err == nil {
 		hash := rawdb.ReadCanonicalHash(db, lastBlock)
-		testhelpers.FailImpl(t, "Didn't fail to get balance at block:", lastBlock, " with hash:", hash, ", lastBlock:", lastBlock)
+		Fatal(t, "Didn't fail to get balance at block:", lastBlock, " with hash:", hash, ", lastBlock:", lastBlock)
 	}
 	if !strings.Contains(err.Error(), "block not found while recreating") {
-		testhelpers.FailImpl(t, "Failed with unexpected error: \"", err, "\", at block:", lastBlock, "lastBlock:", lastBlock)
+		Fatal(t, "Failed with unexpected error: \"", err, "\", at block:", lastBlock, "lastBlock:", lastBlock)
+	}
+}
+
+func testSkippingSavingStateAndRecreatingAfterRestart(t *testing.T, skipBlocks uint32, skipGas uint64, txCount int) {
+	t.Helper()
```

Comment:
this shouldn't be a helper function.
If we get an error here we'll want to see it in the line that failed and not in the line that called this func.